src/string_map.c

```diff
@@ -147,82 +172,52 @@ rcutils_string_map_reserve(rcutils_string_map_t * string_map, size_t capacity)
     return RCUTILS_RET_OK;
   } else if (capacity == 0) {
     // if the requested capacity is zero, then make sure the existing keys and values are free'd
-    allocator.deallocate(string_map->impl->keys, allocator.state);
-    string_map->impl->keys = NULL;
-    allocator.deallocate(string_map->impl->values, allocator.state);
-    string_map->impl->values = NULL;
+    __clear_string_map(string_map);
```

Comment:
This call isn't strictly necessary, because if the size were non-zero, we wouldn't end up at this line. I think I'll remove it, but let me know if you prefer to keep it in just in case, then I'll re-add it.